recurring_contract/models/product_pricelist_item.py

```diff
@@ -0,0 +1,28 @@
+from odoo import models, _
+
+
+class ProductPricelistItem(models.Model):
+    _inherit = 'product.pricelist.item'
+
+    def update_cl_amount(self):
+        for item in self:
+            contract_lines = self.env['recurring.contract.line'].search([
+                ('contract_id.pricelist_id', '=', item.pricelist_id.id),
+                ('product_id', '=', item.product_id.id),
+                ('contract_id.state', '=', ['waiting', 'active'])
```

Comment:
Include also draft sponsorships

Reply:
The draft state is currently not used, when we call the generation of invoices the contract is in waiting state,

Reply:
Yes but if the price is not updated and then a user validates the sponsorship without checking, the sponsorship will not have the same amount as it should when invoice generation will start.